lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2245,7 +2245,17 @@ SubExprInfo ExprEmitter::visitTypeOp(ArraySliceOp op) {
 SubExprInfo ExprEmitter::visitTypeOp(ArrayGetOp op) {
   emitSubExpr(op.getInput(), Selection);
   os << '[';
-  emitSubExpr(op.getIndex(), LowestPrecedence);
+  if (isZeroBitType(op.getIndex().getType())) {
```

Comment:
Should this be here, or should we require that it be pruned away? This one in particular could be tricky to prune, so I'm ok either way.

Reply:
Until we decide on where/when/how to do i0 pruning, then this is necessary (as is for all other `i0` logic which ExportVerilog already has good support for) - got to start i0 support somewhere.

Reply:
This implementation makes sense to me. I guess array-indexing is only valid use of i0 value so we cannot completely prune i0 values at PruneZeroValue pass.